core/textile/bucket_factory.go

```diff
@@ -53,6 +53,22 @@ func getThreadName(userPubKey []byte, bucketSlug string) string {
 	return hex.EncodeToString(userPubKey) + "-" + bucketSlug
 }
 
+func (tc *textileClient) getRemoteBucketContext(ctx context.Context, sDbID string, bucketSlug string) (context.Context, error) {
+	log.Debug("getRemoteBucketContext: Getting remote bucket context")
+
+	dbID, err := utils.ParseDbIDFromString(sDbID)
+	if err != nil {
+		log.Error("Error casting thread id", err)
+		return nil, err
+	}
+	ctx, err = tc.getThreadContext(ctx, bucketSlug, *dbID, true)
```

Comment:
Feels a bit weird that this has a different behaviour than `getBucketContext`, in the sense that the latter creates a bucket if it's not already present and adds it to the metathread collection. Should we instead add a flag to that method to tell it if it should use the hub context or not when calling `getThreadContext`?

Reply:
That works too.  I noticed that difference but wanted to keep this one light instead of all the logic to create it if it doesn't exist because we won't have this one in the collection the same way either so that piece needs to account for this particular bucket not being in the collections.  Maybe we can go with this design to keep it clean:
- Rename my new function to an inner `getBucketContext` that just simply tries to create the context with the `hub` `bool` like you said (I could use this for my use case)
- Turn the existing getBucketContext to something like getOrCreateBucketContext that calls the inner one from 1

Let me know what you think

Reply:
Sounds good to me. We should update the `getOrCreateBucketContext` to internally use your new function then instead of calling `tc.getThreadContext` directly.

Reply:
Fixed in [b842036](https://github.com/FleekHQ/space-daemon/pull/147/commits/b8420364f85811c6db50e8bb634d4e1d948b9a3f).